loopy/kernel/data.py

```diff
@@ -453,13 +452,18 @@ def get_arg_decl(self, ast_builder, name_suffix, shape, dtype, is_written):
 
 class ValueArg(KernelArgument):
     def __init__(self, name, dtype=None, approximately=1000, target=None,
-            is_output_only=False):
+            is_output_only=False, tags=None):
+        """
+        :arg tags: A metadata tag or list of metadata tags intended for
+            consumption by an application. It is intended these tags be
+            instances of :class:`pytools.tag.Tag`.
+        """
```

Comment:
`ValueArg` should inherit from `pytools.tag.Taggable`, which needs to be created.